kafka-consumer-application/kafka/README.md

```diff
@@ -0,0 +1,160 @@
+    }
+  } finally {
+    consumer.close();
+  }
+}
+```
+
+The `FileWritingRecordsHandler` implementation iterates over the `ConsumerRecords` - which contains a collection of
+`ConsumerRecord` objects. Each `ConsumerRecord` object encapsulates metadata and data for the events consumed from Kafka.
+This includes the key, value, topic, partition, headers, offset, timestamp, and many others.
+
+`FileWritingRecordsHandler` writes the `value` of each `ConsumerRecord` to a specified file name.
+
+## Using This Example
+
+<details>
+<summary>Explore the Unit Tests</summary>
+
+There are JUnit test cases in this repo to exercise the `Consumer` functionality.
+
+Clone the `confluentinc/tutorials` GitHub repository (if you haven't already) and navigate to the `tutorials` directory:
+
+```shell
+git clone git@github.com:confluentinc/tutorials.git
+cd tutorials
+```
+
+To run the unit tests, use the provided Gradle Wrapper:
+
+```shell
+./gradlew clean :kafka-consumer-application:kafka:test --info  
+```
+
+The results of the tests can be found in the `build/reports/index.html` report:
+
+![Test Report](./images/test-report.png)
+
+</details>
+
+<details>
+<summary>Execute the Application</summary>
+
+This sample application will consume records from a Kafka topic and write them to a given file. You can run the 
+example application in this tutorial using `confluent local`.
+
+### Prerequisites
+* [Confluent CLI](https://docs.confluent.io/confluent-cli/current/install.html)
+* Docker running via [Docker Desktop](https://docs.docker.com/desktop/) or [Docker Engine](https://docs.docker.com/engine/install/)
+
+### Start Kafka
+
+* Execute `confluent local kafka start`  from a terminal window, and copy the `host:port` output.
+```shell
+The local commands are intended for a single-node development environment only, NOT for production usage. See more: https://docs.confluent.io/current/cli/index.html
+
+
+Pulling from confluentinc/confluent-local
+Digest: sha256:30763749f746295175d6c20b21495fd369b57ca3685175075763865fb6292f6f
+Status: Image is up to date for confluentinc/confluent-local:latest
++-----------------+-------+
+| Kafka REST Port | 8082  |
+| Plaintext Ports | 50277 |
++-----------------+-------+
+Started Confluent Local containers "9cec8b1127".
+To continue your Confluent Local experience, run `confluent local kafka topic create <topic>` and `confluent local kafka topic produce <topic>`.
+```
+
+### Build Application
+
+* Use the Gradle Wrapper provided to build the application.
+```shell
+./gradlew :kafka-consumer-application:kafka:shadowJar
+```
+
+### Execute
+
+* Our application expects 3 input parameters:
+    * The Kafka broker `host:port` - per the `confluent local` step.
+    * Consumer Group ID
+    * Path to an output file, used by the `ConsumerRecordsHandler` implementation.
+
+```shell
+java -jar kafka-consumer-application/kafka/build/libs/kafka-consumer-application-standalone-0.0.1.jar localhost:50277 consumer1 ~/tmp/consumer.out  
```

Comment:
When I run this example I get this
```
Exception in thread "main" java.lang.RuntimeException: java.nio.file.NoSuchFileException: /Users/bill/tmp/consumer.out
        at io.confluent.developer.FileWritingRecordsHandler.process(FileWritingRecordsHandler.java:27)
        at io.confluent.developer.KafkaConsumerApplication.runConsume(KafkaConsumerApplication.java:29)
        at io.confluent.developer.KafkaConsumerApplication.main(KafkaConsumerApplication.java:68)
Caused by: java.nio.file.NoSuchFileException: /Users/bill/tmp/consumer.out
```
If you want, you could have the records print out to the console - up to you.

Reply:
I toyed around with the console a bit. I need to figure out how to make the logs from kafka quiet. I actually like that better than files. I'll refactor it.

Reply:
Done @bbejeck. Refactored to only use the console. Pretty sure I updated all of the corresponding docs as well.